runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -649,12 +818,24 @@ function LanguageTree:included_regions()
     return { {} }
   end
 
-  local regions = {} ---@type Range6[][]
+  local regions = {} ---@type table<integer, Range6[]>
+  local regions_inv = {} ---@type table<integer, (Range6[]|integer)[]>
   for i, _ in pairs(self._trees) do
-    regions[i] = self._trees[i]:included_ranges(true)
+    local region = self._trees[i]:included_ranges(true)
+    prune_empty_ranges(region)
```

Comment:
This needs explaining. Why is treesitter giving back empty ranges? Is that a bug?

Reply:
This happens when the user edits a region in a way that deletes a range in it.

Consider the following markdown text.  
```
* 1
  2
```
Initially, the markdown_inline parser has a single region `{ { 0, 2, 2, 1, 0, 4 }, { 1, 2, 6, 1, 3, 7 } }`.

If you delete the second line, it becomes `{ { 0, 2, 2, 1, 0, 4 }, { 1, 0, 4, 1, 0, 4 } }`. This means that `tree:edit()` doesn't delete empty ranges itself. 

Maybe this could be considered a bug in treesitter.

Reply:
EIther way, this is useful information that should be included in the code.